src/compiler/binder.ts

```diff
@@ -1931,7 +1934,11 @@ namespace ts {
 
                 case SyntaxKind.SpreadElementExpression:
                 case SyntaxKind.JsxSpreadAttribute:
-                    emitFlags |= NodeFlags.HasSpreadAttribute;
+                    let root = container;
+                    while (root && root.kind !== SyntaxKind.BinaryExpression) {
+                        root = root.parent;
+                    }
+                    emitFlags |= root && isDestructuringAssignment(root) ? NodeFlags.HasRestAttribute : NodeFlags.HasSpreadAttribute;

```

Comment:
I'm not 100% sure this is right. You might be walking up from the *right* side of a destructuring assignment, in which case the spread is actually a spread.